From 488abac5bd282b0762f71bc3a9882cfbaa6e2078 Mon Sep 17 00:00:00 2001
From: Alec Ari <neotheuser@ymail.com>
Date: Tue, 23 Jul 2019 19:42:43 -0500
Subject: Fix build with new RTAI on Linux 4.14.134

Needs LinuxCNC commit 469d3cb9b767a48d149b4ab2a02fe49fb07eae26
pci_8255: Change function names to avoid clash with v4.x kernel macros.

Signed-off-by: Alec Ari <neotheuser@ymail.com>
---
 src/configure.ac       | 11 ++---------
 src/rtapi/rtai_rtapi.c |  2 +-
 2 files changed, 3 insertions(+), 10 deletions(-)

diff --git a/src/configure.ac b/src/configure.ac
index 3d0f621a6..7005fcf63 100644
--- a/src/configure.ac
+++ b/src/configure.ac
@@ -371,6 +371,7 @@ case $RTS in
     RTPREFIX=rtai
     RTAI_VERSION=$($RTS --version)
     RTAI=$(echo $RTAI_VERSION | cut -f 1 -d .)
+    RTAI_LINUXVER=`$RTS --full-linux-version`
     RTDIR=`$RTS --prefix`
     RTDIR=$(cd $RTDIR ; pwd -P )
     RTFLAGS=`$RTS --module-cflags`
@@ -593,15 +594,7 @@ install with "sudo apt-get install libusb-1.0-0-dev" or disable with
 
 if test $RTS '!=' uspace; then
 AC_MSG_CHECKING([for kernel version string])
-
-if test -e $KERNELDIR/include/linux/utsrelease.h; then
-    VERSION_FILE=$KERNELDIR/include/linux/utsrelease.h
-elif test -e $KERNELDIR/include/generated/utsrelease.h; then
-    VERSION_FILE=$KERNELDIR/include/generated/utsrelease.h
-else
-    VERSION_FILE=$KERNELDIR/include/linux/version.h
-fi
-KERNEL_VERS=`$CC -E -dM ${VERSION_FILE} | grep UTS | cut -d '"' -f 2`
+KERNEL_VERS="${RTAI_LINUXVER}"
 
 if test -z "$KERNEL_VERS"; then 
     AC_MSG_ERROR(Kernel version string not found)
diff --git a/src/rtapi/rtai_rtapi.c b/src/rtapi/rtai_rtapi.c
index 5733d73d4..1ae8d310e 100644
--- a/src/rtapi/rtai_rtapi.c
+++ b/src/rtapi/rtai_rtapi.c
@@ -64,7 +64,7 @@
 #include <linux/kernel.h>
 #include <linux/slab.h>		/* replaces malloc.h in recent kernels */
 #include <linux/ctype.h>	/* isdigit */
-#include <asm/uaccess.h>	/* copy_from_user() */
+// #include <asm/uaccess.h>	/* copy_from_user() */
 #include <asm/msr.h>		/* rdtscll() */
 
 /* get inb(), outb(), ioperm() */
-- 
2.22.0

